app/lib/azul_search_service.rb

```diff
@@ -201,4 +201,37 @@ def self.extract_file_information(result)
     end
     file_information
   end
+
+  # query Azul with project shortnames and return summary file information
+  # this method mirrors interface on BulkDownloadService#study_download_info for use in bulk download modal
+  def self.get_file_summary_info(accessions)
+    client = ApplicationController.hca_azul_client
+    file_summaries = []
+    file_query = { 'project' => { 'is' => accessions } }
+    results = client.projects(query: file_query)
+    results['hits'].each do |entry|
+      entry_hash = entry.with_indifferent_access
+      project_hash = entry_hash[:projects].first # there will only ever be one project here
+      accession = project_hash[:projectShortname]
+      result = {
+        study_source: 'HCA',
+        name: project_hash[:projectTitle],
+        accession: accession,
+        description: project_hash[:projectDescription],
+        studyFiles: [
+          {
+            project_id: project_hash[:projectId],
+            file_type: 'Project Manifest',
+            count: 1,
+            upload_file_size: 1.megabyte, # placeholder filesize as we don't know until manifest is downloaded
+            name: "#{accession}.tsv"
+          }
+        ]
+      }.with_indifferent_access
+      project_file_info = extract_file_information(entry_hash)
+      result[:studyFiles] += project_file_info if project_file_info.any?
+      file_summaries << result
```

Comment:
```suggestion
     result
```
if using `map` as suggested above

Reply:
39fe8e0